### ISSUE TITLE
Add configuration support for common project customizations that should not require making a host app editable.

### ISSUE THREAD

**Issue by matthew-carroll** [flutter/flutter#22832]
Currently blocked on information gathering, but feel free to work on individual controls that we know we want.

Add configuration support for common project customizations that should not require making a host app editable.

Examples:
 * Firebase plugin configuration
 * Generic API key setup (Maps, Fabric, etc)
 * Permissions
 * Platform API versions
 * Launch icons
 * Splash color, splash image, fullscreen configuration
 * etc

**Comment by MarkOSullivan94** [comment]
Related PR for the Flutter Launcher Icons: https://github.com/fluttercommunity/flutter_launcher_icons/pull/51

Curious on how to proceed with the new changes to the directory structure. I want to be able to support both those who don't have the `.ios` and `.android` directories and also those who do.

**Comment by matthew-carroll** [comment]
@MarkOSullivan94 there may be some changes coming to project structure that permanently places host apps in /android /ios and only includes "module" related structure in .android and .ios.  Please keep an eye out for those changes in the coming week or so and see how that impacts your work.

**Comment by matthew-carroll** [comment]
Closing for now as "won't do".  For now we're going to move forward with a single project structure that always has editable host apps and also includes module linking capabilities.
